packages/orchestration/src/examples/evm-account-kit.js

```diff
@@ -0,0 +1,327 @@
+  send: M.call(M.any(), M.any()).returns(M.any()),
+  sendGmp: M.call(M.any(), M.any()).returns(M.any()),
+  fundLCA: M.call(M.any(), M.any()).returns(VowShape),
+});
+
+const InvitationMakerI = M.interface('invitationMaker', {
+  makeEVMTransactionInvitation: M.call(M.string(), M.array()).returns(M.any()),
+});
+
+const EvmKitStateShape = {
+  localChainAddress: CosmosChainAddressShape,
+  sourceChannel: M.string(),
+  remoteDenom: M.string(),
+  localDenom: M.string(),
+  localAccount: M.remotable('OrchestrationAccount<{chainId:"agoric-3"}>'),
+  assets: M.any(),
+  remoteChainInfo: M.any(),
+};
+harden(EvmKitStateShape);
+
+/**
+ * @param {Zone} zone
+ * @param {{
+ *   zcf: ZCF;
+ *   vowTools: VowTools;
+ *   log: (msg: string) => Vow<void>;
+ *   zoeTools: ZoeTools;
+ * }} powers
+ */
+export const prepareEvmAccountKit = (
+  zone,
+  { zcf, vowTools, log, zoeTools },
+) => {
+  return zone.exoClassKit(
+    'EvmTapKit',
+    {
+      tap: M.interface('EvmTap', {
+        receiveUpcall: M.call(M.record()).returns(
+          M.or(VowShape, M.undefined()),
+        ),
+      }),
+      transferWatcher: M.interface('TransferWatcher', {
+        onFulfilled: M.call(M.undefined())
+          .optional(M.bigint())
+          .returns(VowShape),
+      }),
+      holder: EVMI,
+      invitationMakers: InvitationMakerI,
+    },
+    /** @param {EvmTapState} initialState */
+    initialState => {
+      mustMatch(initialState, EvmKitStateShape);
+      return harden({ evmAccountAddress: undefined, ...initialState });
+    },
+    {
+      tap: {
+        /**
+         * @param {VTransferIBCEvent} event
+         */
+        receiveUpcall(event) {
+          trace('receiveUpcall', event);
+
+          const tx = /** @type {FungibleTokenPacketData} */ (
+            JSON.parse(atob(event.packet.data))
+          );
+
+          trace('receiveUpcall packet data', tx);
+          const memo = JSON.parse(tx.memo);
+
+          if (memo.source_chain === 'Ethereum') {
+            const payloadBytes = decodeBase64(memo.payload);
+            const payload = Array.from(payloadBytes);
+            const decoded = defaultAbiCoder.decode(['address'], payload);
+            trace(decoded);
+
+            this.state.evmAccountAddress = decoded[0];
+          }
+
+          trace('receiveUpcall completed');
+        },
+      },
+      transferWatcher: {
+        /**
+         * @param {void} _result
+         * @param {bigint} value the qty of uatom to delegate
+         */
+        onFulfilled(_result, value) {
+          trace('onFulfilled _result:', JSON.stringify(_result));
+          trace('onFulfilled value:', JSON.stringify(value));
+          trace('onFulfilled state:', JSON.stringify(this.state));
+        },
+      },
+      holder: {
+        async getAddress() {
+          // @ts-expect-error
+          const localChainAddress = await this.state.localAccount.getAddress();
+          return localChainAddress.value;
+        },
```

Comment:
Since this is outside a flow, we can't use async/await.

Also, the `ERef<OrchestrationAccount>` at the top is wanting you to do `E(this.state.localAccount)`.

This object is held locally in your vat, so we don't need `E`:

```suggestion
       getLocalAddress() {
          return this.state.localAccount.getAddress();
        },
```